reference/preconditioner/sor_kernels.cpp

```diff
@@ -0,0 +1,70 @@
+// SPDX-FileCopyrightText: 2017 - 2024 The Ginkgo authors
+//
+// SPDX-License-Identifier: BSD-3-Clause
+
+#include "core/preconditioner/sor_kernels.hpp"
+
+#include <memory>
+
+#include <ginkgo/core/base/math.hpp>
+#include <ginkgo/core/matrix/csr.hpp>
+
+#include "reference/factorization/factorization_helpers.hpp"
+
+
+namespace gko {
+namespace kernels {
+namespace reference {
+namespace sor {
+
+
+template <typename ValueType, typename IndexType>
+void initialize_weighted_l(
+    std::shared_ptr<const DefaultExecutor> exec,
+    const matrix::Csr<ValueType, IndexType>* system_matrix,
+    remove_complex<ValueType> weight, matrix::Csr<ValueType, IndexType>* l_mtx)
+{
+    auto inv_weight = one(weight) / weight;
+    factorization::helpers::initialize_l(
+        system_matrix, l_mtx,
+        factorization::helpers::triangular_mtx_closure(
+            [inv_weight](auto val) { return val * inv_weight; },
+            [](auto val) { return val; }));
```

Comment:
```suggestion
            factorization::helpers::identity{}));
```
Couldn't you use the identity here?